client/app/pages/queries/QuerySource.jsx

```diff
@@ -41,6 +42,98 @@ function chooseDataSourceId(dataSourceIds, availableDataSources) {
   return find(dataSourceIds, id => includes(availableDataSources, id)) || null;
 }
 
+function confirmOverwrite() {
+  return new Promise((resolve, reject) => {
+    Modal.confirm({
+      title: "Overwrite Query",
+      content: (
+        <React.Fragment>
+          <div className="m-b-5">It seems like the query has been modified by another user.</div>
+          <div>Are you sure you want to overwrite the query with your version?</div>
+        </React.Fragment>
+      ),
+      okText: "Overwrite",
+      okType: "danger",
+      onOk: () => {
+        resolve();
+      },
+      onCancel: () => {
+        reject();
+      },
+      maskClosable: true,
+      autoFocusButton: null,
+    });
+  });
+}
+
+function saveQuery(query, data, options) {
```

Comment:
These methods will be shared with the QueryView, so perhaps have them in a separate file?

Reply:
Of course :slightly_smiling_face: Already doing that :+1:

Reply:
Done